quality-assurance/test-code-standards.md

```diff
@@ -1,49 +1,73 @@
+### .NET (back-end projects)
+* For .NET projects the standard test framework is [NUnit](https://docs.nunit.org/articles/nunit/intro.html), and the preferred mocking library is [FakeItEasy](https://fakeiteasy.github.io/docs/7.4.0/).
+* Although not mandated, we strongly advise using the [Fluent Assertions](https://fluentassertions.com/introduction) package for test assertions. Fluent Assertions provides clearer failure messages and greater capacity for object and enumerable assertions.
+* It is advised to use [WebApplicationFactory](https://learn.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-7.0) for API tests, see API Test Standards (link tbc) for more information.
 
-## API Testing
+###  Angular (front-end projects)
+* For Angular projects the standard unit test framework is [Jest](https://jestjs.io/docs/getting-started).
+* For UI tests the suggested UI framework is [Playwright](https://playwright.dev/docs/intro) (typescript). See [Browser automation standards](browser-automation.md) for more information.
 
-* Perform full happy/unhappy path tests at this level
-* Interactions with APIs should be abstracted into a separate service/facade, not alongside the test code
+### Other technologies
+* Although not mandated using a mutation testing tool such as [Stryker](https://stryker-mutator.io/docs/stryker-net/introduction/) will help give an objective score regarding the quality of the unit tests.
+* The approved performance testing technology is [K6](https://k6.io/docs/). Please see performance testing standards (link tbc) for further information.
 
-## UI Testing
+## Standards
 
-* Only use for e2e tests or explicit UI features at this level
-* Interactions with UIs should be abstracted into a separate service/facade, not alongside the test code (e.g., Page Object Model)
+### Unit Testing
+#### Test a single Method in isolation
+* Test naming should follow a convention agreed by the team. An example is **Method_Action_Assertion** e.g. ValidateObject_WhenObjectPropteryIsNull_ThrowsArgurmentException
+* Tests should be isolated, i.e. have no dependencies on other tests nor on order of execution.
+* Unit tests should take very little time to run. Milliseconds not seconds!
+* Unit tests are standalone, can be run in isolation, and have no dependencies on any outside factors such as a file system or database.
+* Running a unit test should be consistent with its results, it always returns the same outcome if you do not change anything in between runs.
+* The test should be able to automatically detect if it passed or failed without any human interaction.
+* A unit test should not take a disproportionately long time to write compared to the code being tested. If you find testing the code taking a large amount of time compared to writing the code, consider a design that is more testable.
+* Unit Test coverage should aim to cover around 80% of code.
```

Comment:
they should focus on the most complex functionality in order to ensure these are tested thoroughly